src/VisualStudio/Core/Def/Implementation/Debugging/AbstractBreakpointResolver.cs

```diff
@@ -87,59 +84,72 @@ private BreakpointResolutionResult CreateBreakpoint(ISymbol methodSymbol)
-                // We only expect to hit this condition when the name provided does not parse.
-                return SpecializedCollections.EmptyList<ISymbol>();
-            }
-            else if (nameParts.Count == 1)
-            {
-                // They're just searching for a method name.  Have to look through every type to find
-                // it.
-                return FindMembers(await GetAllTypesAsync(cancellationToken).ConfigureAwait(false), nameParts[0]);
-            }
-            else if (nameParts.Count == 2)
-            {
-                // They have a type name and a method name.  Find a type with a matching name and a
-                // method in that type.
-                var types = await GetAllTypesAsync(cancellationToken).ConfigureAwait(false);
-                types = types.Where(t => MatchesName(t, nameParts[0], _identifierComparer));
-                return FindMembers(types, nameParts[1]);
+                switch (nameParts.Count)
+                {
+                    case 0:
+                        // If there were no name parts, then we don't have any members to return.
+                        // We only expect to hit this condition when the name provided does not parse.
+                        return SpecializedCollections.EmptyList<ISymbol>();
+
+                    case 1:
+                        // They're just searching for a method name.  Have to look through every type to find
+                        // it.
+                        return FindMembers(await GetAllTypesAsync(cancellationToken).ConfigureAwait(false), nameParts[0]);
+
+                    case 2:
+                        // They have a type name and a method name.  Find a type with a matching name and a
+                        // method in that type.
+                        var types = await GetAllTypesAsync(cancellationToken).ConfigureAwait(false);
+                        types = types.Where(t => MatchesName(t, nameParts[0], _identifierComparer));
+                        return FindMembers(types, nameParts[1]);
+
+                    default:
+                        // They have a namespace or nested type qualified name.  Walk up to the root namespace trying to match.
+                        var containers = await _solution.GetGlobalNamespacesAsync(cancellationToken).ConfigureAwait(false);
+                        return FindMembers(containers, nameParts.ToArray());
+                }
             }
-            else
+            catch (Exception e) when (FatalError.ReportWithoutCrashUnlessCanceled(e))
             {
-                // They have a namespace or nested type qualified name.  Walk up to the root namespace trying to match.
-                var containers = await _solution.GetGlobalNamespacesAsync(cancellationToken).ConfigureAwait(false);
-                return FindMembers(containers, nameParts.ToArray());
+                return ImmutableArray<ISymbol>.Empty;
```

Comment:
Same argument as above - no state is modified by this code and the caller already needs to deal with empty result.